restart_plugin/mtcp_restart_plugin.c

```diff
@@ -849,6 +849,12 @@ mtcp_plugin_skip_memory_region_munmap(Area *area, RestoreInfo *rinfo)
   LhCoreRegions_t *lh_regions_list = NULL;
   int total_lh_regions = lh_info->numCoreRegions;
 
+  // Don't skip munmap of mtcp_restart regions.
+  if (mtcp_strendswith(area->name, "/mtcp_restart") ||
+      mtcp_strendswith(area->name, "[heap]")) {
```

Comment:
we only want to skip the [heap] right after mtcp_restart. Also in mpi_plugin.cpp we need to do the same to skip those area for libsStart consideration.